src/js/components/returns/inbound/AddItemsPage.jsx

```diff
@@ -396,22 +394,34 @@ class AddItemsPage extends Component {
     const itemsWithLotAndExpirationDate = returnItems.filter(
       (it) => it.expirationDate && it.lotNumber,
     );
-
+    const itemsWithMismatchedExpiry = [];
     // Trying to find at least one instance where the data that we are trying to save
     // does not match the expiration date of the existing inventoryItem in the system
     // eslint-disable-next-line no-restricted-syntax
-    for (const it of itemsWithLotAndExpirationDate) {
-      // eslint-disable-next-line no-await-in-loop
-      const { data } = await ProductApi.getInventoryItem(it.product?.id, it.lotNumber);
-      if (data.inventoryItem && data.inventoryItem.expirationDate !== it.expirationDate) {
-        // After finding at least a single instance where expiration date we are trying to save
-        // does not match the existing inventoryItem expiration date, we want to inform the user
-        // that certain updates to th expiration date in the system will be performed
-        // eslint-disable-next-line no-await-in-loop
-        const shouldUpdateLotExpirationDate = await this.confirmExpirationDateSave();
-        if (!shouldUpdateLotExpirationDate) {
-          return Promise.reject();
+    await Promise.all(
+      itemsWithLotAndExpirationDate.map(async (it) => {
+        const { data } = await ProductApi.getInventoryItem(it.product?.id, it.lotNumber);
+        if (data.inventoryItem && data.inventoryItem.expirationDate !== it.expirationDate) {
+          return itemsWithMismatchedExpiry.push({
+            code: it.product?.productCode,
+            product: it.product,
+            lotNumber: it.lotNumber,
+            previousExpiry: data.inventoryItem.expirationDate,
+            newExpiry: it.expirationDate,
+          });
         }
+        return null;
+      }),
+    );
+
+    if (itemsWithMismatchedExpiry.length > 0) {
+      // After finding at least a single instance where expiration date we are trying to save
+      // does not match the existing inventoryItem expiration date, we want to inform the user
+      // that certain updates to th expiration date in the system will be performed
+      const shouldUpdateLotExpirationDate =
```

Comment:
decide on the naming, either use the `*expirationDate` everywhere only or `*lotExpirationDate`